python/postprocessing/examples/collectionMerger.py

```diff
@@ -0,0 +1,77 @@
+
+class collectionMerger(Module):
+
+    def __init__(self,input,output,sortkey = lambda x : x.pt,reverse=True,selector=None,maxObjects=None):
+        self.input = input
+        self.output = output
+        self.nInputs = len(self.input)
+        self.sortkey = lambda (obj,j,i) : sortkey(obj)
+        self.reverse = reverse
+        self.selector = selector # pass lambda function to be called on each object
+        self.maxObjects = maxObjects # save only the first maxObjects objects passing the selection in the merged collection
+        self.branchType = {}
+        self.nullValue = []
+        pass
+    def beginJob(self):
+        pass
+    def endJob(self):
+        pass
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        _brlist = inputTree.GetListOfBranches()
+        branches = [_brlist.At(i) for i in xrange(_brlist.GetEntries())]
+        self.brlist_sep = [self.filterBranchNames(branches,x) for x in self.input]
+        self.brlist_all = set(itertools.chain(*(self.brlist_sep)))
+
+        self.is_there = np.zeros(shape=(len(self.brlist_all),self.nInputs),dtype=bool)
+        for bridx,br in enumerate(self.brlist_all):
+            for j in xrange(self.nInputs):
+                if br in self.brlist_sep[j]: self.is_there[bridx][j]=True
+
+        self.nullValue = [ 0.0 if self.branchType[br] in ['Float_t','Double_t'] else 0 for br in self.brlist_all]
+
+        self.out = wrappedOutputTree
+        for br in self.brlist_all:
+            self.out.branch("%s_%s"%(self.output,br), _rootLeafType2rootBranchType[self.branchType[br]], lenVar="n%s"%self.output)
+
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def filterBranchNames(self,branches,collection):
+        out = []
+        for br in branches:
+            name = br.GetName()
+            if not re.match(collection+'_.*',name): continue
+            out.append(name.replace(collection+'_',''))
+            self.branchType[out[-1]] = br.FindLeaf(br.GetName()).GetTypeName()
+        return out
+
+    def analyze(self, event):
+        """process event, return True (go to next module) or False (fail, go to next event)"""
+        coll = [Collection(event,x) for x in self.input]
+        objects = [(coll[j][i],j,i) for j in xrange(self.nInputs) for i in xrange(len(coll[j]))]
+        if self.selector: objects=filter(lambda (obj,j,i) : self.selector(obj), objects)
+        objects.sort(key = self.sortkey, reverse = self.reverse)
+        if self.maxObjects: objects = objects[:self.maxObjects]
+        for bridx,br in enumerate(self.brlist_all):
+            out = []
+            nullvalue = self.nullValue[bridx]
+            for obj,j,i in objects:
+                val = getattr(obj,br) if self.is_there[bridx][j] else nullvalue
+                if type(val)==str: val = ord(val)
+                out.append(val)
```

Comment:
could be worth checking if using a pre-allocated array of ints or floats will be better than doing append in a python list